libmscore/read302.cpp

```diff
@@ -187,7 +187,7 @@ bool Score::read(XmlReader& e)
                   else {
                         e.tracks().clear();     // ???
                         MasterScore* m = masterScore();
-                        Score* s       = new Score(m, MScore::baseStyle());
+                        Score* s       = new Score(m, m->style());
```

Comment:
This seems to be what is responsible for https://musescore.org/en/node/316559.

If I understand what is going on here, we are about to read the part.  It may its own style settings we read from the file, but for everything else, we should be taking the defaults from the "regular" defaults.  Instead, here we are taking them from the parent score, leading to exactly the problem described.

This change was made, I assume, to make sure that whatever "keep old style" setting we applied to the score also make their way into the part.  Better, then, to do this directly.  That is, don't look to the parent score, but the what set of style defaults the parent score was itself drawing from.  We've got a whole bunch of functions available, I assume one of them does the trick.

Reply:
Seems to me the same code used in setting style for the main score should be used here.  So, I stole/adapted this from MasterScore::read1() and it *seems* to work, replacing the line above with the following:

                        Score* s       = new Score(m, MScore::baseStyle());
                        int defaultsVersion = m->style().defaultStyleVersion();
                        s->setStyle(*MStyle::resolveStyleDefaults(defaultsVersion));
                        s->style().setDefaultStyleVersion(defaultsVersion);

Could be there is some detail I have wrong, or something more elegant.  Anyhow, this is, I think, the basic right idea.  The excerpt should have it's style defaults set in the same way the master score's style defaults were set - so the same "keep old style" settings apply.  As opposed to what is happening now, which is actually copy the master score's current style settings.

Preusmably the same sort of change would happening in read302.